wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -0,0 +1,874 @@
+                                                         metadata=edit_role_binding_name)
+        auth_api.create_namespaced_role_binding(namespace=proj_name, body=edit_role_binding)
+
+        self.logger.info("project sa created via api have no some mandatory roles. adding them")
+        self._restore_missing_project_role_bindings(namespace=proj_name)
+
+        # creating pods and etc
+        processing_params = {'DATABASE_PASSWORD': password,
+                             'APPLICATION_DOMAIN': proj_url}
+        processing_params.update(prepared_tags)
+        self.logger.info(("processing template and passed params in order to "
+                          "prepare list of required project entities"))
+        template_entities = self.process_template(name=template, namespace=self.default_namespace,
+                                                  parameters=processing_params)
+        self.logger.debug("template entities:\n {e}".format(e=template_entities))
+        kinds = set([e['kind'] for e in template_entities])
+        entity_names = {e: inflection.underscore(e) for e in kinds}
+        proc_names = {k: 'create_{e}'.format(e=p) for k, p in entity_names.items()}
+        for entity in template_entities:
+            if entity['kind'] in kinds:
+                procedure = getattr(self, proc_names[entity['kind']], None)
+                # todo: this code should be paralleled
+                obtained_entity = procedure(namespace=proj_name, **entity)
+                self.logger.debug(obtained_entity)
+            else:
+                self.logger.error("some entity %s isn't present in entity creation list", entity)
+
+        # obtaining db ip
+        common_svc = self.k_api.read_namespaced_service(name='common-service',
+                                                        namespace=proj_name)
+        ext_ip = common_svc.spec.external_i_ps[0]
+
+        self.logger.info("verifying that all created entities are up and running")
+        try:
+            wait_for(self.is_vm_running, num_sec=600,
+                     func_kwargs={'vm_name': proj_name})
+            logger.info("all pods look up and running")
+            return {'url': proj_url,
+                    'external_ip': ext_ip,
+                    'project': proj_name,
+                    }
+        except TimedOutError:
+            logger.error("deployment failed. Please check failed pods details")
+            # todo: return and print all failed pod details
+            return None
+
+    def start_vm(self, vm_name):
+        """Starts a vm.
+
+        Args:
+            vm_name: name of the vm to be started
+        Returns: whether vm action has been initiated properly
+        """
+        self.logger.info("starting vm/project %s", vm_name)
+        if self.does_project_exist(vm_name):
+            for pod in self.required_project_pods:
+                self.scale_entity(name=pod, namespace=vm_name, replicas=1)
```

Comment:
**QUESTION**: I wonder what if you had a project with pods having 2 replicas, this would scale them down, right?

Reply:
exactly, this value is passed to replication controller and it decides whether it should increase or decrease number of pods.
I also thought of this case and I decided to take care of such possible cases when we start testing such cases.

Reply:
I was thinking that it would only scale up pods that have 0 replicas

Reply:
nope, I thought user should take care of this. I can add such check if you this is it necessary